poly1305_amd64.go

```diff
@@ -10,6 +10,11 @@ import (
 	"io"
 )
 
+const (
+	AVX2Size   = 512
```

Comment:
Maybe don't export the constant, clients should not know anything about the internal implementation...